src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -52,37 +52,54 @@ public async Task PrepareJobAsync(IExecutionContext context, List<ContainerInfo>
-            jobContainer.ContainerId = response.Context.Container.Id;
-            context.JobContext.Container["network"] = new StringContextData(response.Context.Container.Network);
-            jobContainer.ContainerNetwork = response.Context.Container.Network;
-            context.JobContext["hook_state"] = new StringContextData(JsonUtility.ToString(response.State));
-
-            // TODO: figure out if we need ContainerRuntimePath for anything
-            // var configEnvFormat = "--format \"{{range .Config.Env}}{{println .}}{{end}}\"";
-            // var containerEnv = await _dockerManager.DockerInspect(executionContext, container.ContainerId, configEnvFormat);
-            // container.ContainerRuntimePath = DockerUtil.ParsePathFromConfigEnv(containerEnv);
-
-            for (var i = 0; i < response.Context.Services.Count; i++)
+            if (response.Context != null)
             {
-                var container = response.Context.Services[i]; // TODO: Confirm that the order response.Context.Services is the same as serviceContainers
-                var containerInfo = serviceContainers[i];
-                containerInfo.ContainerId = container.Id;
-                containerInfo.ContainerNetwork = container.Network;
-                var service = new DictionaryContextData()
+                if (response.Context.Container != null)
                 {
-                    ["id"] = new StringContextData(container.Id),
-                    ["ports"] = new DictionaryContextData(),
-                    ["network"] = new StringContextData(container.Network)
-                };
-
-                // TODO: workout port mappings + format
-                // foreach (var portMapping in containerInfo.UserPortMappings)
-                // {
-                //     // TODO: currently the format is ports["80:8080"] = "80:8080", fix this?
-                //     (service["ports"] as DictionaryContextData)[$"{portMapping.Key}:{portMapping.Value}"] = new StringContextData($"{portMapping.Key}:{portMapping.Value}");
-                // }
-                context.JobContext.Services[containerInfo.ContainerNetworkAlias] = service;
+                    if (response.Context.Container.Id != null)
+                    {
+                        context.JobContext.Container["id"] = new StringContextData(response.Context.Container.Id);
+                        jobContainer.ContainerId = response.Context.Container.Id;
+                    }
+                    
+                    if (response.Context.Container.Network != null)
+                    {
+                        context.JobContext.Container["network"] = new StringContextData(response.Context.Container.Network);
+                        jobContainer.ContainerNetwork = response.Context.Container.Network;
+                    }
+
+                    context.JobContext["hook_state"] = new StringContextData(JsonUtility.ToString(response.State));
```

Comment:
I don't think this should be in this if statement, but we may want to wrap it in a 

```
if (response.State)
```
block